beetsplug/musicbrainz.py

```diff
@@ -458,6 +481,63 @@ def __init__(self):
                 "'musicbrainz.search_limit'",
             )
 
+    def _apply_pseudo_release_config(self):
+        self._scripts = self.config["pseudo_releases"]["scripts"].as_str_seq()
+        self._log.debug("Desired pseudo-release scripts: {0}", self._scripts)
+
+        album_custom_tags = (
+            self.config["pseudo_releases"]["album_custom_tags"].get().keys()
+        )
+        track_custom_tags = (
+            self.config["pseudo_releases"]["track_custom_tags"].get().keys()
+        )
+        self._log.debug(
+            "Custom tags for albums and tracks: {0} + {1}",
+            album_custom_tags,
+            track_custom_tags,
+        )
+        for custom_tag in album_custom_tags | track_custom_tags:
+            if not isinstance(custom_tag, str):
+                continue
+
+            media_field = mediafile.MediaField(
+                mediafile.MP3DescStorageStyle(custom_tag),
+                mediafile.MP4StorageStyle(
+                    f"----:com.apple.iTunes:{custom_tag}"
+                ),
+                mediafile.StorageStyle(custom_tag),
+                mediafile.ASFStorageStyle(custom_tag),
+            )
+            try:
+                self.add_media_field(custom_tag, media_field)
+            except ValueError:
+                # ignore errors due to duplicates
+                pass
+
+        self.register_listener(
+            "album_info_received", self._determine_pseudo_album_info_ref
+        )
+        self.register_listener("album_matched", self._adjust_final_album_match)
+
+    def _determine_pseudo_album_info_ref(
+        self,
+        items: Iterable[Item],
+        album_info: AlbumInfo,
+    ):
+        if isinstance(album_info, PseudoAlbumInfo):
+            for item in items:
+                # particularly relevant for reimport but could also happen during import
+                if "mb_albumid" in item:
+                    del item["mb_albumid"]
+                if "mb_trackid" in item:
+                    del item["mb_trackid"]
```

Comment:
If you reimport because you want to switch from an official release to a pseudo-release, the MB ids from the official release introduce a lot of penalty here, that's why I delete them. I think this is fine as long as it is not flushed to disk/DB, right?